src/utils/components/DiskModal/utils/helpers.ts

```diff
@@ -138,12 +140,19 @@ export const getDataVolumeFromState = (
         url: `docker://${diskSourceState.registrySource}`,
       },
     };
-  } else {
+  } else if (diskState.diskSource === sourceTypes.HTTP) {
     dataVolume.spec.source = {
       [diskState.diskSource]: {
         url: diskSourceState.urlSource,
       },
     };
+  } else if (diskState.diskSource === sourceTypes.DATA_SOURCE) {
+    dataVolume.spec.sourceRef = {
+      kind: DataSourceModel.kind,
+      name: diskSourceState.dataSourceName,
+      namespace: diskSourceState.dataSourceNamespace,
+    };
+    delete dataVolume?.spec?.source;
```

Comment:
isn't it better to remove https://github.com/kubevirt-ui/kubevirt-plugin/blob/main/src/utils/components/DiskModal/utils/helpers.ts#L36 ?

Reply:
you're right, great tip :)